tensorflow/tools/ci_build/builds/test_installation.sh

```diff
@@ -0,0 +1,293 @@
+#
+# tensorflow/python/framework/ops_test.py
+#   depends on depends on "test_ops", which is defined in a C++ file wrapped as
+#   a .py file through the Bazel rule “tf_gen_ops_wrapper_py”.
+# tensorflow/util/protobuf/compare_test.py:
+#   depends on compare_test_pb2 defined outside Python
+# tensorflow/python/framework/device_test.py:
+#   depends on CheckValid() and ToString(), both defined externally
+#
+PY_TEST_BLACKLIST="${PY_TEST_BLACKLIST}:"\
+"tensorflow/python/framework/ops_test.py:"\
+"tensorflow/python/util/protobuf/compare_test.py:"\
+"tensorflow/python/framework/device_test.py"
+
+# Test blacklist: GPU-only
+PY_TEST_GPU_BLACKLIST="${PY_TEST_GPU_BLACKLIST}:"\
+"tensorflow/python/framework/function_test.py"
+
+# =============================================================================
+
+
+# Helper functions
+# Get the absolute path from a path
+abs_path() {
+  [[ $1 = /* ]] && echo "$1" || echo "$PWD/${1#./}"
+}
+
+
+die() {
+  echo $@
+  exit 1
+}
+
+
+# Obtain the path to Python binary
+# source tools/python_bin_path.sh
+if [[ "$1" == "--virtualenv" ]]; then
+  PYTHON_BIN_PATH="$(which python)"
+else
+  source tools/python_bin_path.sh
+  # Assume: PYTHON_BIN_PATH is exported by the script above
+fi
+
+if [[ -z "${PYTHON_BIN_PATH}" ]]; then
+  die "PYTHON_BIN_PATH was not provided. If this is not virtualenv, "\
+"did you run configure?"
+fi
+
+# Determine the major and minor versions of Python being used (e.g., 2.7)
+# This info will be useful for determining the directory of the local pip
+# installation of Python
+PY_MAJOR_MINOR_VER=$(${PYTHON_BIN_PATH} -V 2>&1 | awk '{print $NF}' | cut -d. -f-2)
+
+echo "Python binary path to be used in PIP install-test: ${PYTHON_BIN_PATH} "\
+"(Major.Minor version: ${PY_MAJOR_MINOR_VER})"
+
+
+# Directory from which the unit-test files will be run
+PY_TEST_DIR_REL="pip_test/tests"
+PY_TEST_DIR=$(abs_path ${PY_TEST_DIR_REL})  # Get absolute path
+rm -rf ${PY_TEST_DIR} && mkdir -p ${PY_TEST_DIR}
+
+# Create test log directory
+PY_TEST_LOG_DIR_REL=${PY_TEST_DIR_REL}/logs
+PY_TEST_LOG_DIR=$(abs_path ${PY_TEST_LOG_DIR_REL})  # Absolute path
+
+mkdir ${PY_TEST_LOG_DIR}
+
+
+# Copy source files that are required by the tests but are not included in the
+# PIP package
+
+# Look for local Python library directory
+LIB_PYTHON_DIR=""
+
+pushd /tmp > /dev/null
+TF_FILE=$("${PYTHON_BIN_PATH}" -c "import tensorflow as tf; print(tf.__file__)") \
+|| die "FAILED: Unable to import tensorflow in python: ${PYTHON_BIN_PATH}"
+popd > /dev/null
+
+# A typical value of TF_FILE would be:
+#   /usr/local/lib/python2.7/dist-packages/tensorflow/__init__.pyc
+LIB_PYTHON_DIR=$(dirname $(dirname $(dirname "${TF_FILE}")))
+
+if [[ -z ${LIB_PYTHON_DIR} ]]; then
+  die "Failed to find Python library directory where TensorFlow is installed."
+else
+  echo "Found Python library directory where TensorFlow is installed: "\
+"${LIB_PYTHON_DIR}"
+fi
+
+PACKAGES_DIR=$(ls -d ${LIB_PYTHON_DIR}/*-packages | head -1)
```

Comment:
Why are the 20 lines above and bellow so complicated? The PACKAGES_DIR computation currently looks fragile. Would this work?
TF_INSTALLATION_PATH=$(dirname $("${PYTHON_BIN_PATH}" -c "import tensorflow as tf; print(tf.__file__)"))

Note: No need of PACKAGES_DIR in the first place as TF_INSTALLATION_PATH=${PACKAGES_DIR}/tensorflow

And everything from pushd can be gone.